### PR TITLE
Challenges: Add capability for pseudo remote evaluation

### PR DESCRIPTION
**Pull request by deshraj**
This PR adds the following capabilities:

- [x] Ability to do remote evaluation where the submission worker sends the evaluation job to other API endpoint hosted by challenge organizer
- [x] Update the Challenge model to support `remote_evaluation` as boolean field
- [x] Update ZipChallengeSerializer 
- [x] Update Challenge creation docs for the new field

### CODE REVIEW

File: scripts/workers/submission_worker.py [hunk 1]
```diff
@@ -288,24 +288,34 @@ def run_submission(challenge_id, challenge_phase, submission, user_annotation_fi
...
     # main directory does not gets polluted
     temp_run_dir = join(submission_data_dir, 'run')
     create_dir(temp_run_dir)
 
-    stdout_file_name = 'temp_stdout.txt'
-    stderr_file_name = 'temp_stderr.txt'
-
-    stdout_file = join(temp_run_dir, stdout_file_name)
-    stderr_file = join(temp_run_dir, stderr_file_name)
+    stdout_file = join(temp_run_dir, 'temp_stdout.txt')
```
Comment (RishabhJain2018) [inline scripts/workers/submission_worker.py]: Just to confirm, we are making this change so as to remove the unnecessary variables `stdout_file_name`, `stderr_file_name`, right?

File: scripts/workers/submission_worker.py [hunk 2]
```diff
@@ -519,7 +529,6 @@ def main():
 
     while True:
         for message in queue.receive_messages():
-            # Print out the body of the message
```
Comment (RishabhJain2018) [inline scripts/workers/submission_worker.py]: Why we are removing this comment in this PR?

File: scripts/workers/submission_worker.py (hunk 1 above)
Comment (deshraj) [reply scripts/workers/submission_worker.py]: Yep. These variables were defined for no reason.

File: scripts/workers/submission_worker.py (hunk 2 above)
Comment (deshraj) [reply scripts/workers/submission_worker.py]: This comment is pretty much useless. You can directly figure out what is going on in that line and don't need this comment.

Comment (RishabhJain2018) [reply scripts/workers/submission_worker.py]: I know, just wanted to highlight it so that we won't wonder about it in future.
Rest Looks goo!

**Comment by deshraj** [comment]
> @deshraj I think we also need to update the challenge creation using zip file API, if we are updating the documentation as the challenge hosts will try to add that key and challenge creation will fail.

I think we just need to add the field of `remote_evaluation` to the `ZipChallengeSerializer` to make the challenge creation API work with the new field. Isn't it the case? Moreover, since the default value is `False` for this field so it will take care if the field is not supplied in the config file. 

Please let me know if I am missing something here.

**Comment by RishabhJain2018** [comment]
> I think we just need to add the field of remote_evaluation to the ZipChallengeSerializer to make the challenge creation API work with the new field. Isn't it the case?
True.
Looks good!

**Pull request merged**